grakn-engine/src/main/java/ai/grakn/engine/SystemKeyspaceSessionProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine;
+
+import ai.grakn.GraknConfigKey;
+import ai.grakn.GraknTxType;
+import ai.grakn.Keyspace;
+import ai.grakn.factory.EmbeddedGraknSession;
+import ai.grakn.factory.SystemKeyspaceSession;
+import ai.grakn.kb.internal.EmbeddedGraknTx;
+
+/**
+ * Provides transactions to "graknsystem" which is Grakn system keyspace - used by GraknKeyspaceStore to keep track
+ * of available keyspaces
+ *
+ * @author Marco Scoppetta
+ */
+
+public class SystemKeyspaceSessionProvider implements SystemKeyspaceSession {
```

Comment:
Calling this `_Provider` confused me. I thought it was a factory :(

I don't know what a good name is though. `SystemKeyspaceSessionImpl` is a bit silly 🤷‍♂️

Reply:
@aelred, maybe `GraknSystemKeyspaceSession` ?

Reply:
👍 
Makes sense, and that's the convention you used before I think.